packages/common/src/models/USDCTransactions.ts

```diff
@@ -0,0 +1,79 @@
+import { Nullable } from '../utils/typeUtils'
+
+import { StringUSDC } from './Wallet'
+
+export enum USDCTransactionType {
+  PURCHASE = 'PURCHASE',
+  PURCHASE_CONTENT = 'PURCHASE_CONTENT',
+  TRANSFER = 'TRANSFER'
+}
+
+export enum USDCTransactionMethod {
+  // Transfer methods
+  SEND = 'SENT',
+  RECEIVE = 'RECEIVED',
+
+  // Purchase Methods
+  STRIPE = 'STRIPE'
+}
+
+export enum USDCTransactionMetadataType {
+  PURCHASE_USDC = 'PURCHASE_USDC',
+  PURCHASE_CONTENT = 'PURCHASE_CONTENT',
+  TRANSER = 'TRANSFER'
+}
+
+export type InAppUSDCPurchaseMetadata = {
+  discriminator: USDCTransactionMetadataType.PURCHASE_USDC
+  usd: string
+  usdc: StringUSDC
+  purchaseTransactionId: string
+}
+
+export enum USDCContentPurchaseType {
+  TRACK = 'TRACK'
+}
+
+export type USDCPurchaseContentMetadata = {
+  discriminator: USDCTransactionMetadataType.PURCHASE_CONTENT
+  amount: StringUSDC
+  senderUserId: string
+  receiverUserId: string
+  purchaseType: USDCContentPurchaseType
+  contentId: string
```

Comment:
Wonder if we want these all to be string ids or if we want numeric `ID`s? @piazzatron wdyt?

Reply:
Good catch - numeric IDs! Endpoint should return the encoded string ID but internally we always use the integer version

Reply:
Oh nice catch, thank you.